cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -525,44 +570,375 @@ def _migrate_component(
             created=created_at,
             created_by=created_by,
         )
-    if component_existed and not replace_existing:
+
+    # Component existed and we do not replace it and it is not deleted previously
+    if component_existed and not component_deleted and not replace_existing:
         return component.versioning.draft.publishable_entity_version
+
+    # If component existed and was deleted or we have to replace the current version
+    # Create the new component version for it
     component_version = libraries_api.set_library_block_olx(target_key, new_olx_str=olx)
     for filename, content_pk in content_by_filename.items():
         filename_no_ext, _ = os.path.splitext(filename)
         if filename_no_ext not in olx:
             continue
         new_path = f"static/{filename}"
-        try:
-            authoring_api.create_component_version_content(component_version.pk, content_pk, key=new_path)
-        # @@TODO is there a way to determine content already exists that doesn't run the risk
-        #        of swallowing other, unexpected IntegrityErrors?
-        except IntegrityError:
-            pass  # Content already exists
+        _create_component_version_content_safely(
+            component_version_pk=component_version.pk,
+            content_pk=content_pk,
+            key=new_path
+        )
     return component_version.publishable_entity_version
 
 
-def _slugify_source_usage_key(key: UsageKey) -> str:
+def _deduplicate_container(
+    source_key: UsageKey,
+    target_library_key: LibraryLocatorV2,
+    container_type: ContainerType,
+    title: str,
+    existing_source_to_target_keys: dict[UsageKey, PublishableEntityVersion],
+    preserve_url_slugs: bool = False
+) -> LibraryUsageLocatorV2:
```

Comment:
```suggestion
) -> LibraryContainerLocator:
```

Unlike with modulestore, containers are not actually XBlocks in Learning Core, so they don't have usage keys. The LibraryContainerLocator is a separate class.

Reply:
Sure, it's just a typo - the returned value is indeed a **LibraryContainerLocator**. Will update the return type.